hardhat/test/aave-vault.js

```diff
@@ -0,0 +1,186 @@
+      ethers.utils.parseEther('1.01')
+    );
+    const yield = await instance.yield();
+
+    // assert
+    expect(yield).to.equal(ethers.utils.parseEther('0.01'));
+  });
+
+  it('aave - should return yield 0 if no yield', async () => {
+    // arrange
+    const { instance, _mockERC20 } = await setup();
+
+    // act
+    await instance.deposit({
+      value: ethers.utils.parseEther('1'),
+    });
+    _mockERC20.smocked.balanceOf.will.return.with(
+      ethers.utils.parseEther('0.99')
+    );
+    const yield = await instance.yield();
+
+    // assert
+    expect(yield).to.equal(ethers.utils.parseEther('0'));
+  });
+
+  it('aave - can withdraw', async () => {
+    // arrange
+    const { instance, _mockERC20, _mockAaveGateway } = await setup();
+    const [owner, bob, player] = await ethers.getSigners();
+
+    // act
+    await hre.network.provider.request({
+      method: 'hardhat_impersonateAccount',
+      params: [player.address],
+    });
+    _mockERC20.smocked.balanceOf.will.return.with(
+      ethers.utils.parseEther('1.01')
+    );
+    await instance.connect(player).withdraw(bob.address);
+
+    // assert
+    expect(_mockAaveGateway.smocked.withdrawETH.calls[0]._pool).to.eq(POOL);
+    expect(_mockAaveGateway.smocked.withdrawETH.calls[0]._amount).to.eq(
+      ethers.utils.parseEther('1.01')
+    );
+    expect(_mockAaveGateway.smocked.withdrawETH.calls[0]._to).to.eq(
+      bob.address
+    );
+  });
+
+  it('aave - can not withdraw if not player contract', async () => {
+    // arrange
+    const { instance, _mockERC20 } = await setup();
+    const [owner, bob, player, peter] = await ethers.getSigners();
+
+    // act
+    await hre.network.provider.request({
+      method: 'hardhat_impersonateAccount',
+      params: [peter.address],
+    });
+    _mockERC20.smocked.balanceOf.will.return.with(
+      ethers.utils.parseEther('1.01')
+    );
+    await expect(instance.connect(peter).withdraw(bob.address)).to.be.reverted;
+  });
+
+  it('aave - can emergency withdraw', async () => {
+    // arrange
+    const { instance } = await setup();
+    const [owner] = await ethers.getSigners();
+
+    // act
+    await instance.withdrawEmergency();
+    const _owner = await instance.owner();
+
+    // assert
+    expect(_owner).to.eq(owner.address);
+  });
+
+  it('aave - can transfer ownership', async () => {
+    // arrange
+    const { instance } = await setup();
+    const [owner, bob] = await ethers.getSigners();
+
+    // act
+    await instance.transferOwnership(bob.address);
+    const _owner = await instance.owner();
+
+    // assert
+    expect(_owner).to.eq(bob.address);
+  });
+
+  it('aave - second initialize should revert', async () => {
+    // arrange
+    const { instance, _mockAaveGateway, _mockERC20 } = await setup();
+
+    // act
+    await expect(
+      instance.initialize(
+        _mockAaveGateway.address,
+        _mockERC20.address,
+        POOL,
+        POOL
+      )
+    ).to.be.reverted;
```

Comment:
should second 'POOL' not be the player contract address, as per the function definition? (or am I missing something?)

Reply:
yes sure, since for the test its not relevant it does not really matter, should probably use the 0x0000 address instead